app/Decsys/Services/WebhookService.cs

```diff
@@ -55,30 +56,28 @@ public async Task Trigger(PayloadModel payload)
     /// <returns>True if filter matches</returns>
     private static bool FilterCriteria(WebhookModel webhook, PayloadModel payload)
     {
-        foreach (var eventType in webhook.TriggerCriteria.SelectMany(filter => filter.EventTypes))
-        {
-            // check 1: if they are the same event types
-            var result = eventType.GetType() == payload.EventType.GetType();
-
-            if(result)
-                // check 2: type specific validation criteria
-                // add further criteria for further types
-                result = eventType switch {
-                    PageNavigation nav => CheckIsValid(nav, payload.EventType as PageNavigation),
-                    _ => false
-                };
+        // check 1: if they are the same event types
+        var result = payload.EventType is PageNavigation && 
```

Comment:
This check is 1. Not extensible - only checks for PageNavigation and 2. isn't quite right for the webhook config

to fix:

1. check if `webhook.TriggerCriteria.HasCustomTriggers` is `false` - if it's `false` then return `true` because all trigger events are valid for this webhook.
2. switch on `payload.EventType` and for each possible type, call a valid trigger check
    - e.g. if the type is PageNavigation, then call `IsValidPageNavigationTrigger()` and return the result of that check